fix: export lang types for use in custom component

Grisoly opened the pull request:
## Changes

- Exported BundledLanguage, SpecialLanguage, and LanguageRegistration types from Astro's public API.
- Enables users to type-check the lang prop in custom components without workarounds.
- Fixes #14009.

## Testing

- Manually verified that the types can be imported from astro in a test project.
- Confirmed that valid values (e.g., 'ts') pass type checking and invalid values fail.

<!-- DON'T DELETE THIS SECTION! If no tests added, explain why. -->

## Docs

No user-facing docs changes required, but this improves DX for custom code block components.
Consider adding a note in the component API docs about the new type exports.
<!-- If docs will be needed or you’re not sure, uncomment the next line: -->
 /cc @withastro/maintainers-docs for feedback!

<!-- DON'T DELETE THIS SECTION! If no docs added, explain why.-->

- No user-facing documentation changes are needed because this PR only exports internal types for advanced TypeScript usage. The public API and user experience remain unchanged for most users.

<!-- https://github.com/withastro/docs -->

Grisoly commented:
The remaining CI failures appear to be unrelated to my changes (mostly fork/infra issues). Please let me know if you need me to rebase or address anything specific.

florian-lefebvre reviewed (changes requested):
Types are exported from another file. Can you instead do the following?
- Create a new type called `CodeLanguage` in `src/types.d.ts` that is an union of the 3 types you have here
- Remove the file you created
- Update `components/Code.astro` to use this new type

Grisoly commented:
@florian-lefebvre I've implemented the requested changes:

1. Removed direct re-exports (No longer exporting shiki types directly from index.ts)
2. Created `CodeLanguage` union type (Added to `packages/astro/types.d.ts` with all three language types in alphabetical order)
3. Updated `Code.astro`  (that now uses the new 'CodeLanguage` type instead of direct shiki imports)
4. Deleted index.ts file 

The implementation now follows the suggested specifications and gives type-safe access to supported code languages without Typescript suppressions.

florian-lefebvre reviewed (approved):
Thanks!

florian-lefebvre commented on packages/astro/types.d.ts:
```diff
@@ -1,6 +1,7 @@
 import './astro-jsx';
 import type { OmitIndexSignature, Simplify } from './dist/type-utils.js';
 import type { AstroBuiltinAttributes } from './dist/types/public/elements.js';
+import type { BundledLanguage, SpecialLanguage, LanguageRegistration } from 'shiki';
```
I think we need to move this to a file in `src/types/public` instead, I need to investigate
